widgets/SalesforceBatchSource-batchsource.json

```diff
@@ -0,0 +1,90 @@
+{
+  "metadata": {
+    "spec-version": "1.5"
+  },
+  "configuration-groups": [
+    {
+      "label": "Reference",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Reference Name",
+          "name": "referenceName"
+        }
+      ]
+    },
+    {
+      "label": "Authentication",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Username",
+          "name": "username"
+        },
+        {
+          "widget-type": "password",
+          "label": "Password",
+          "name": "password"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Client Id",
+          "name": "clientId"
+        },
+        {
+          "widget-type": "password",
+          "label": "Client Secret",
+          "name": "clientSecret"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Login Url",
+          "name": "loginUrl",
+          "widget-attributes" : {
+            "default": "https://login.salesforce.com/services/oauth2/token"
+          }
+        }
+      ]
+    },
+    {
+      "label": "Query",
+      "properties": [
+        {
+          "widget-type": "textarea",
+          "label": "SOQL Query",
+          "name": "query",
+          "widget-attributes" : {
+            "placeholder": "SELECT <offset field>, ... FROM <object> WHERE <offset field> > <offset> ORDER BY <offset field>"
+          },
+          "plugin-function": {
+            "method": "POST",
+            "widget": "getSchema",
+            "output-property": "schema",
+            "plugin-method": "getSchema"
+          }
+        }
+      ]
+    }
+  ],
+  "outputs": [
+    {
+      "name": "schema",
+      "label": "schema",
+      "widget-type": "schema",
+      "widget-attributes": {
+        "schema-types": [
+          "boolean",
+          "int",
+          "long",
+          "float",
+          "double",
+          "bytes",
+          "string",
+          "map<string, string>"
```

Comment:
1. Do we support maps?
2. Date, timestamp?

Reply:
1. Removing that
2. I just looked at other example. I don't see them specifying logical types here.